src/features/island/hud/components/codex/Codex.tsx

```diff
@@ -37,20 +36,39 @@ import { hasFeatureAccess } from "lib/flags";
 import { ChoreBoard } from "./pages/ChoreBoard";
 import { FLOWERS } from "features/game/types/flowers";
 import { CompetitionDetails } from "features/competition/CompetitionBoard";
+import { MachineState } from "features/game/lib/gameMachine";
 
 interface Props {
   show: boolean;
   onHide: () => void;
 }
 
+const _farmId = (state: MachineState) => state.context.farmId;
+const _state = (state: MachineState) => state.context.state;
+const _username = (state: MachineState) => state.context.state.username;
+const _bertObsession = (state: MachineState) =>
+  state.context.state.bertObsession;
+const _npcs = (state: MachineState) => state.context.state.npcs;
+const _bounties = (state: MachineState) => state.context.state.bounties;
+const _delivery = (state: MachineState) => state.context.state.delivery;
+const _choreBoard = (state: MachineState) => state.context.state.choreBoard;
+const _kingdomChores = (state: MachineState) =>
+  state.context.state.kingdomChores;
+const _faction = (state: MachineState) => state.context.state.faction;
```

Comment:
If you're getting the `_state` above then there is no real benefit in the additional selectors here. You can just grab what you need from `state`. 

I guess the rule is:

If somewhere in the component you need to pass the whole `state` then just create a selector for the `state`. 

If you don't require it then be selective about exactly the values you need.

It also highlights the issue of required props for components or arguments for functions. We should always ask "do we need to pass the whole state?". A lot of the time we just use one value once inside the component/function. Could we just pass that one value instead. 

This comment is not to say you need to change that because it occurs everywhere but its good to think about it in the future.